docs/user/install.rst

```diff
@@ -37,39 +37,68 @@ Or, to install the latest beta or release candidate, if any:
 
     $ pip install --pre falcon
 
-In order to provide an extra speed boost, Falcon can compile itself with
-Cython. Wheels containing pre-compiled binaries are available from PyPI for
-several common platforms (see :ref:`binary_wheels` below for the complete list
-of the platforms that we target, or simply check
+In order to provide an extra speed boost, Falcon automatically compiles itself
+with `Cython <https://cython.org/>`__. Wheels containing pre-compiled binaries
+are available from PyPI for the majority of common platforms (see
+:ref:`binary_wheels` below for the complete list of the platforms that we
+target, or simply check
 `Falcon files on PyPI <https://pypi.org/project/falcon/#files>`__).
 
-However, even if a wheel for your platform of choice is not available, you can
-choose to stick with the source distribution, or use the instructions below to
-cythonize Falcon for your environment.
+However, even if a binary build for your platform of choice is not available,
+you can choose to stick with the generic pure-Python wheel (that ``pip`` should
+pick automatically), or cythonize Falcon for your environment (see
+:ref:`instructions below <cythonize>`).
+The pure-Python version is functionally identical to binary wheels;
+it is just slower on CPython.
 
-The following commands tell pip to install Cython, and then to invoke
-Falcon's ``setup.py``, which will in turn detect the presence of Cython
-and then compile (AKA cythonize) the Falcon framework with the system's
-default C compiler.
+.. _cythonize:
+
+Cythonizing Falcon
+^^^^^^^^^^^^^^^^^^
+
+Falcon leverages `PEP 517 <https://peps.python.org/pep-0517/>`__ to
+automatically compile (AKA *cythonize*) itself with Cython whenever it is
+installed from the source distribution. So if a suitable
+:ref:`binary wheel <binary_wheels>` is unavailable for your platform, or if you
+want to recompile locally, you simply need to instruct ``pip`` not to use
+prebuilt wheels:
 
 .. code:: bash
 
-    $ pip install cython
-    $ pip install --no-build-isolation --no-binary :all: falcon
+    $ pip install --no-binary :all: falcon
 
-Note that ``--no-build-isolation`` is necessary to override pip's default
-PEP 517 behavior that can cause Cython not to be found in the build
-environment.
+If you want to verify that Cython is being invoked,
+pass ``-v`` to ``pip`` in order to echo the compilation commands:
+
+.. code:: bash
 
-If you want to verify that Cython is being invoked, simply
-pass `-v` to pip in order to echo the compilation commands:
+    $ pip install -v --no-binary :all: falcon
+
+Apart from the obvious requirement to have a functional compiler toolchain set
+up with CPython development headers, the only inconvenience of running
+cythonization on your side is the extra couple of minutes it takes (depending
+on your hardware; it can take much more on an underpower CI runner, or if you
```

Comment:
```suggestion
on your hardware; it can take much more on an underpowered CI runner, or if you
```

Reply:
Thanks, didn't proofread it myself somehow...